### DISCUSSION TITLE
How to get list of all routes

### DISCUSSION THREAD

**Discussion by andrecasal** [remix-run/remix#9280] [Q&A]
Is it possible to get a list of all existing routes (outside of a route/inside a Playwright test)?

**Comment by jrestall** [comment]
There's a lot of properties on the window object that could be useful. Usual disclaimer being that these are internal and could break in any update.
```js
window.__remixRouter
window.__remixRouteModules
window.__remixManifest
```
Once you have the router object you can do lots of things if you are okay using private/internal methods:
```ts
const routes = window.__remixRouter.routes;
```
